File: frontend/src/services/incidentService.js
import api from './api';

export const incidentService = {
  createIncident: async (incidentData) => {
    const res = await api.post('/incidents', incidentData);
    return res?.data || res;
  },

  getIncidents: async (params = {}) => {
    const res = await api.get('/incidents', { params });
    return res?.data || res;
  },

  getActiveIncidents: async () => {
    const res = await api.get('/incidents/active');
    return res?.data || res;
  },

  getDashboardStats: async () => {
    const res = await api.get('/incidents/stats');
    return res?.data || res;
  },

  getIncidentById: async (id) => {
    const res = await api.get(`/incidents/${id}`);
    return res?.data || res;
  },

  getIncidentByNumber: async (incidentNumber) => {
    const res = await api.get(`/incidents/number/${incidentNumber}`);
    return res?.data || res;
  },

  updateStatus: async (id, statusData) => {
    const res = await api.patch(`/incidents/${id}/status`, statusData);
    return res?.data || res;
  },

  assignOfficer: async (id, assignmentData) => {
    const res = await api.patch(`/incidents/${id}/assign`, assignmentData);
    return res?.data || res;
  },

  cancelIncident: async (id) => {
    const res = await api.patch(`/incidents/${id}/cancel`);
    return res?.data || res;
  },

  getIncidentHistory: async (id) => {
    const res = await api.get(`/incidents/${id}/history`);
    return res?.data || res;
  },
};

File: backend/src/main/java/com/campusguard/service/IncidentService.java
package com.campusguard.service;

import com.campusguard.dto.*;
import com.campusguard.entity.User;
import com.campusguard.enums.IncidentStatus;
import com.campusguard.enums.IncidentType;
import com.campusguard.enums.Severity;
import org.springframework.data.domain.Pageable;

import java.util.List;

public interface IncidentService {
    IncidentResponse createIncident(IncidentCreateRequest request, User reporter);
    IncidentResponse getIncidentById(Long id, User currentUser);
    IncidentResponse getIncidentByNumber(String incidentNumber, User currentUser);
    
    PageResponse<IncidentResponse> getIncidents(
            IncidentStatus status,
            Severity severity,
            IncidentType type,
            Long locationId,
            String search,
            User currentUser,
            Pageable pageable
    );

    List<IncidentResponse> getActiveIncidents();

    IncidentResponse updateStatus(Long id, IncidentStatusRequest request, User currentUser);
    IncidentResponse assignOfficer(Long id, IncidentAssignmentRequest request, User currentUser);
    
    List<IncidentHistoryResponse> getIncidentHistory(Long incidentId, User currentUser);

    DashboardStatsResponse getDashboardStats();

    void cancelIncident(Long id, User student);
}
